File: RustSideStory/SampleSource/enums.swift
// Some people say Swift Enums inspired by Rust

enum Language {
	case rust
	case swift
	case other(name: String)

	var awesome: Bool {
		switch self {
		case .rust: return true
		case .swift: return true
		case .other(let name): return name == "Go"
		}
	}
}

print("\(Language.rust.awesome) " +
	"\(Language.swift.awesome) " +
	"\(Language.other(name: "Go").awesome) " +
	"\(Language.other(name: "ObjC").awesome)")

File: RustSideStory/SampleSource/enums.rs
// Rust Enums are awesome! Can contain values!

enum Language {
    Rust,
    Swift,
    Other(String),
}

impl Language {
    fn awesome(self) -> bool {
        match self {
            Language::Rust => return true,
            Language::Swift => return true,
            Language::Other(name) => return name == "Go",
        }
    }
}

fn main() {
    println!("{:?} {:?} {:?} {:?}",
             Language::Rust.awesome(),
             Language::Swift.awesome(),
             Language::Other("Go".to_string()).awesome(),
             Language::Other("ObjC".to_string()).awesome());
}
